src/charts/Line.js

```diff
@@ -372,7 +372,11 @@ class Line {
           /* fix #374 */
           sX = w.globals.seriesX[realIndex][iterations - 1]
         }
-        x = (sX - w.globals.minX) / this.xRatio
+        if (isNull && w.config.chart.spanNullValues) {
+          x = x
```

Comment:
can we remove this condition to remove this line?

Reply:
If you mean `if (isNull && w.config.chart.spanNullValues)`: I had to add it, to get the curves between `null` values right, but maybe you can optimize/remove it.

Reply:
No worries, I will optimize it to avoid `x = x` statement.
Thanks

Reply:
Think I misunderstood you. Of course this can be refactored into something more elegantly, like:

```javascript
if (w.globals.isXNumeric) {
  let sX = w.globals.seriesX[realIndex][j + 1]
  if (typeof w.globals.seriesX[realIndex][j + 1] === 'undefined') {
    sX = w.globals.seriesX[realIndex][iterations - 1]
  }
  if (!(isNull && w.config.chart.spanNullValues)) {
    x = (sX - w.globals.minX) / this.xRatio
  } 
} else {
  x = x + this.xDivision
}
```

Hope this is what you meant.